<<<<<<<< HEAD:package/python3/0007-Add-an-option-to-disable-lib2to3.patch
From d53e271be8952fd95a593741f91bf0ef2a7e5f02 Mon Sep 17 00:00:00 2001
========
From 18e19402fab3939223a58481734e685d25c3102b Mon Sep 17 00:00:00 2001
>>>>>>>> upstream/master:package/python3/0009-Add-an-option-to-disable-lib2to3.patch
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 22 Feb 2017 17:15:31 -0800
Subject: [PATCH] Add an option to disable lib2to3

lib2to3 is a library to convert Python 2.x code to Python 3.x. As
such, it is probably not very useful on embedded system targets.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Samuel Martin <s.martin49@gmail.com>
[ Andrey Smirnov: ported to Python 3.6 ]
Signed-off-by: Andrey Smirnov <andrew.smirnov@gmail.com>
[ Adam Duskett: ported to Python 3.10.0 ]
Signed-off-by: Adam Duskett <aduskett@gmail.com>
[ Bernd Kuhls: ported to Python 3.11.4]
Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
[ Adam Duskett: ported to Python 3.12.1 ]
Signed-off-by: Adam Duskett <adam.duskett@amarulasolutions.com>
---
 Makefile.pre.in | 19 ++++++++++++++-----
 configure.ac    |  6 ++++++
 2 files changed, 20 insertions(+), 5 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
<<<<<<<< HEAD:package/python3/0007-Add-an-option-to-disable-lib2to3.patch
index ecf1b995c01..a5feddcbbc3 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -2052,7 +2052,9 @@ ifeq (@PYDOC@,yes)
========
index e87bc1fcd32..1a0ab0de426 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1881,7 +1881,9 @@ ifeq (@PYDOC@,yes)
>>>>>>>> upstream/master:package/python3/0009-Add-an-option-to-disable-lib2to3.patch
 	(cd $(DESTDIR)$(BINDIR); $(LN) -s pydoc$(VERSION) pydoc3)
 endif
 	-rm -f $(DESTDIR)$(BINDIR)/2to3
+ifeq (@LIB2TO3@,yes)
 	(cd $(DESTDIR)$(BINDIR); $(LN) -s 2to3-$(VERSION) 2to3)
+endif
 	if test "x$(LIPO_32BIT_FLAGS)" != "x" ; then \
 		rm -f $(DESTDIR)$(BINDIR)/python3-32$(EXE); \
 		(cd $(DESTDIR)$(BINDIR); $(LN) -s python$(VERSION)-32$(EXE) python3-32$(EXE)) \
<<<<<<<< HEAD:package/python3/0007-Add-an-option-to-disable-lib2to3.patch
@@ -2099,7 +2101,6 @@ LIBSUBDIRS=	asyncio \
========
@@ -1927,7 +1929,6 @@ LIBSUBDIRS=	asyncio \
>>>>>>>> upstream/master:package/python3/0009-Add-an-option-to-disable-lib2to3.patch
 		idlelib idlelib/Icons \
 		importlib importlib/resources importlib/metadata \
 		json \
-		lib2to3 lib2to3/fixes lib2to3/pgen2 \
 		logging \
 		multiprocessing multiprocessing/dummy \
 		re \
<<<<<<<< HEAD:package/python3/0007-Add-an-option-to-disable-lib2to3.patch
@@ -2203,10 +2204,6 @@ TESTSUBDIRS=	idlelib/idle_test \
 		test/test_importlib/resources/zipdata02 \
 		test/test_importlib/source \
 		test/test_json \
-		test/test_lib2to3 \
-		test/test_lib2to3/data \
-		test/test_lib2to3/data/fixers \
-		test/test_lib2to3/data/fixers/myfixes \
 		test/test_module \
 		test/test_peg_generator \
 		test/test_sqlite3 \
@@ -2253,6 +2250,14 @@ ifeq (@PYDOC@,yes)
========
@@ -1947,10 +1948,6 @@ LIBSUBDIRS=	asyncio \
 TESTSUBDIRS=	ctypes/test \
 		distutils/tests \
 		idlelib/idle_test \
-		lib2to3/tests \
-		lib2to3/tests/data \
-		lib2to3/tests/data/fixers \
-		lib2to3/tests/data/fixers/myfixes \
 		test \
 		test/audiodata \
 		test/certdata \
@@ -2074,6 +2071,14 @@ ifeq (@PYDOC@,yes)
>>>>>>>> upstream/master:package/python3/0009-Add-an-option-to-disable-lib2to3.patch
 LIBSUBDIRS += pydoc_data
 endif
 
+ifeq (@LIB2TO3@,yes)
+LIBSUBDIRS += lib2to3 lib2to3/fixes lib2to3/pgen2
+TESTSUBDIRS += test/test_lib2to3			\
+		test/test_lib2to3/data \
+		test/test_lib2to3/data/fixers \
+		test/test_lib2to3/data/fixers/myfixes
+endif
+
 TEST_MODULES=@TEST_MODULES@
<<<<<<<< HEAD:package/python3/0007-Add-an-option-to-disable-lib2to3.patch
 
 .PHONY: libinstall
@@ -2336,10 +2341,12 @@ ifeq (@PYC_BUILD@,yes)
 		-o 0 -o 1 -o 2 $(COMPILEALL_OPTS) -d $(LIBDEST)/site-packages -f \
========
 libinstall:	all $(srcdir)/Modules/xxmodule.c
 	@for i in $(SCRIPTDIR) $(LIBDEST); \
@@ -2176,10 +2181,12 @@ ifeq (@PYC_BUILD@,yes)
 		-j0 -d $(LIBDEST)/site-packages -f \
>>>>>>>> upstream/master:package/python3/0009-Add-an-option-to-disable-lib2to3.patch
 		-x badsyntax $(DESTDIR)$(LIBDEST)/site-packages
 endif
+ifeq (@LIB2TO3@,yes)
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -m lib2to3.pgen2.driver $(DESTDIR)$(LIBDEST)/lib2to3/Grammar.txt
 	-PYTHONPATH=$(DESTDIR)$(LIBDEST) $(RUNSHARED) \
 		$(PYTHON_FOR_BUILD) -m lib2to3.pgen2.driver $(DESTDIR)$(LIBDEST)/lib2to3/PatternGrammar.txt
+endif
 
 # bpo-21536: Misc/python-config.sh is generated in the build directory
 # from $(srcdir)Misc/python-config.sh.in.
@@ -2465,7 +2472,9 @@ libainstall: all scripts
 	$(INSTALL_SCRIPT) $(srcdir)/install-sh $(DESTDIR)$(LIBPL)/install-sh
 	$(INSTALL_SCRIPT) python-config.py $(DESTDIR)$(LIBPL)/python-config.py
 	$(INSTALL_SCRIPT) python-config $(DESTDIR)$(BINDIR)/python$(LDVERSION)-config
+ifeq (@LIB2TO3@,yes)
 	$(INSTALL_SCRIPT) $(SCRIPT_2TO3) $(DESTDIR)$(BINDIR)/2to3-$(VERSION)
+endif
 	$(INSTALL_SCRIPT) $(SCRIPT_IDLE) $(DESTDIR)$(BINDIR)/idle$(VERSION)
 ifeq (@PYDOC@,yes)
 	$(INSTALL_SCRIPT) $(SCRIPT_PYDOC) $(DESTDIR)$(BINDIR)/pydoc$(VERSION)
diff --git a/configure.ac b/configure.ac
<<<<<<<< HEAD:package/python3/0007-Add-an-option-to-disable-lib2to3.patch
index 277a16a903e..654f903ef12 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7535,6 +7535,12 @@ PY_STDLIB_MOD([xxlimited_35], [test "$with_trace_refs" = "no"], [test "$ac_cv_fu
========
index d56d8f9d1c5..bf91fb2f824 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7147,6 +7147,12 @@ PY_STDLIB_MOD([xxlimited_35], [test "$with_trace_refs" = "no"], [test "$ac_cv_fu
>>>>>>>> upstream/master:package/python3/0009-Add-an-option-to-disable-lib2to3.patch
 # substitute multiline block, must come after last PY_STDLIB_MOD()
 AC_SUBST([MODULE_BLOCK])
 
+AC_SUBST(LIB2TO3)
+
+AC_ARG_ENABLE(lib2to3,
+	AS_HELP_STRING([--disable-lib2to3], [disable lib2to3]),
+	[ LIB2TO3="${enableval}" ], [ LIB2TO3=yes ])
+
 # generate output files
<<<<<<<< HEAD:package/python3/0007-Add-an-option-to-disable-lib2to3.patch
 AC_CONFIG_FILES(m4_normalize([
   Makefile.pre
-- 
2.43.0
========
 AC_CONFIG_FILES(Makefile.pre Misc/python.pc Misc/python-embed.pc Misc/python-config.sh)
 AC_CONFIG_FILES([Modules/Setup.bootstrap Modules/Setup.stdlib])
diff --git a/setup.py b/setup.py
index d8647724650..e02591dae10 100644
--- a/setup.py
+++ b/setup.py
@@ -1598,11 +1598,11 @@ class DummyProcess:
     import warnings
     warnings.filterwarnings("ignore",category=DeprecationWarning)
 
-    scripts = ['Tools/scripts/idle3', 'Tools/scripts/2to3',
-               'Lib/smtpd.py']
+    scripts = ['Tools/scripts/idle3', 'Lib/smtpd.py']
     if not '--disable-pydoc' in sysconfig.get_config_var("CONFIG_ARGS"):
         scripts += [ 'Tools/scripts/pydoc3' ]
-
+    if not '--disable-lib2to3' in sysconfig.get_config_var("CONFIG_ARGS"):
+        scripts += [ 'Tools/scripts/2to3' ]
     setup(# PyPI Metadata (PEP 301)
           name = "Python",
           version = sys.version.split()[0],
-- 
2.44.0
>>>>>>>> upstream/master:package/python3/0009-Add-an-option-to-disable-lib2to3.patch

